### PR TITLE
 feat(toggleOpen): #102 - Change ToggleOpen to not be a Button 

### PR DESCRIPTION
**Pull request by JoelJKoel**
To solve the issue #102. 

The problem with these changes is that old `ToggleOpen` supported all `Button` props, with this change only some of them are implemented. Should more optional props be supported and is there a better way to add them?

Solves the accessibility problem and should look and handle the same way in given examples.
image

### CODE REVIEW

**Comment by kasparsuvi1** [comment]
I dont think this is the way to go:
1. We are duplicating button styles
2. When ToggleOpen is wanted to use as button (main use-case), it cannot be used liked that.

For example we could use directly `ButtonContent` element, and pass `as` element as div or button depending on the useCase.
Or use  `Button` component inside ToggleOpen and add `as` property to Toggleopen, that can be changed to `div` needed. Im not 100% sure if with the second approach we should also override type="button" from WCAG point of view or is it acceptable and valid HTML.

**Comment by JoelJKoel** [comment]
Thanks for the feedback! Seems like adding an `as` prop to render the button as a `div` when needed was indeed enough to fix the WCAG accessibility issue.

image

File: libs/react-components/community/src/components/toggle-open/toggle-open.tsx [hunk 1]
```diff
@@ -23,20 +23,35 @@ export interface ToggleOpenProps extends Omit<ButtonProps, 'children' | 'iconRig
    * Overidden from ButtonProps, because we dont support string as icon name here
    */
   iconRight?: Partial<IconProps>;
+  /**
+   * If the element is a button or a div
+   * @default button
+   */
+  as?: 'button' | 'div';
```
Comment (kasparsuvi1) [inline libs/react-components/community/src/components/toggle-open/toggle-open.tsx:30]: PolymorphicComponentPropWithRef types should probaly used here, thought nothing is broken with it.
Poly props are not very straightforward to use, but Sidenav for example has example of it.
@airikej is it ok?

**Review by kasparsuvi1 — APPROVED** [review]

File: libs/react-components/community/src/components/toggle-open/toggle-open.tsx (hunk 1 above)
Comment (airikej) [reply libs/react-components/community/src/components/toggle-open/toggle-open.tsx:30]: Yes I think its fine for now, ready to merge!

**Review by airikej — APPROVED** [review]

**Comment by kasparsuvi1** [comment]
@JoelJKoel had to revert after merge, because the build did not complete. #231

**Pull request merged**